src/dev_server.rs

```diff
@@ -9,26 +9,77 @@ use std::{
     process,
 };
 
+/// A simple HTTP server that can, optionally, watch a given command.
+///
+/// Get the files at `watch_path` and serve them at a given IP address
+/// (127.0.0.1:8000 by default). An optional command can be run along the
+/// server, relaunched if changes are detected in your project via a [`Watch`].
+///
+/// # Usage
+///
+/// ```rust,no_run
+/// # use std::process;
+/// # use xtask_wasm::{anyhow::Result, clap};
+/// #
+/// # #[derive(clap::Parser)]
+/// # struct Opt {
+/// #    #[clap(subcommand)]
+/// #    cmd: Command,
+/// # }
+/// #
+/// #[derive(clap::Parser)]
+/// enum Command {
+///     Serve(xtask_wasm::DevServer),
+/// }
+///
+/// fn main() -> Result<()> {
+///     let opt: Opt = clap::Parser::parse();
+///
+///     match opt.cmd {
+///         Command::Serve(mut dev_server) => {
+///             let mut command = process::Command::new("cargo");
+///             command.args(["xtask", "dist"]);
+///
+///             dev_server.watch = dev_server.watch.exclude_workspace_path("dist");
+///
+///             println!("Starting to serve at {}:{}", dev_server.ip, dev_server.port);
+///             dev_server.command(command).start("dist")?;
+///         }
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// Add a `serve` subcommand that will run `cargo xtask dist`,
+/// watching for changes in the workspace and serve the files in the `dist`
+/// directory at a given IP address.
 #[non_exhaustive]
 #[derive(Debug, Parser)]
 pub struct DevServer {
+    /// Ip address to bind. Default to `127.0.0.1`
     #[clap(long, default_value = "127.0.0.1")]
     pub ip: IpAddr,
+    /// Port number. Default to `8000`
     #[clap(long, default_value = "8000")]
     pub port: u16,
 
-    #[clap(flatten)]
-    pub watch: Watch,
+    /// Watched command running along the server
```

Comment:
I don't understand the meaning of this sentence tbh :thinking: 
```suggestion
    /// Command executed when a change is detected.
    ///
    /// Use `None` to not watch for changes.
```